daemon/qrexec-daemon-common.c

```diff
@@ -135,7 +135,8 @@ bool qrexec_execute_vm(const char *target, bool autostart, int remote_domain_id,
             return false;
         }
     }
-    if (autostart) {
+    int s = connect_unix_socket(target);
+    if (s == -2 && autostart) {
```

Comment:
This should only be done if the target VM is not a disposable VM.  If the target VM is a fresly created disposable VM, it will definitely not be running here, so the check is pointless.
```suggestion
    int s = disposable ? -2 : connect_unix_socket(target);
    if (s == -2 && autostart) {
```